SPDX-FileCopyrightText: 2025 fosslinux <fosslinux@aussies.space>
SPDX-FileCopyrightText: 2012 Karl Williamson <public@khwilliamson.com>

SPDX-License-Identifier: Artistic-1.0

This symbol goes unused in this commit, but the next version of perl
that we build changed the name of the symbol, so rename it here by
backporting this patch.

Only the part of the patch changing the name is used. The other two
modifications in the original patch change the use of the symbol
which obviously does not apply to this build.

From b79624bd711790796036524d5fc46521233cf85f Mon Sep 17 00:00:00 2001
From: Karl Williamson <public@khwilliamson.com>
Date: Sun, 16 Dec 2012 11:50:34 -0700
Subject: [PATCH] Rename property involved in \X matching, for clarity

I was re-reading some code and got confused.  This table matches just
the first character of a sequence that may or may not contain others.
---
 lib/unicore/mktables  | 8 ++++----
 3 files changed, 9 insertions(+), 9 deletions(-)

diff --git perl-5.17.4/lib/unicore/mktables perl-5.17.4/lib/unicore/mktables
index e779b08e8e..b9dc1df.17 100644
--- perl-5.17.4/lib/unicore/mktables
+++ perl-5.17.4/lib/unicore/mktables
@@ -13551,7 +13551,7 @@ sub compile_perl() {
         push @tables_that_may_be_empty, $ri->full_name;
     }
 
-    my $specials_begin = $perl->add_match_table('_X_Special_Begin',
+    my $specials_begin = $perl->add_match_table('_X_Special_Begin_Start',
                                        Perl_Extension => 1,
                                        Fate => $INTERNAL_ONLY,
                                        Initialize => $lv_lvt_v
@@ -13560,9 +13560,9 @@ sub compile_perl() {
                                                    + $ri
                                       );
     $specials_begin->add_comment(join_lines( <<END
-For use in \\X; matches first character of potential multi-character
-sequences that can begin an extended grapheme cluster.  They need special
-handling because of their complicated nature.
+For use in \\X; matches first (perhaps only) character of potential
+multi-character sequences that can begin an extended grapheme cluster.  They
+need special handling because of their complicated nature.
 END
     ));
     my $regular_begin = $perl->add_match_table('_X_Regular_Begin',
-- 
2.49.1

